compiler/vm.nim

```diff
@@ -1511,10 +1511,11 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       regs[ra].intVal = ord(
         # Note that `nfIsRef` + `nkNilLit` represents an allocated
         # reference with the value `nil`, so `isNil` should be false!
-        (node.kind == nkNilLit and nfIsRef notin node.flags) or
-        (not node.typ.isNil and node.typ.kind == tyProc and
-          node.typ.callConv == ccClosure and node[0].kind == nkNilLit and
-          node[1].kind == nkNilLit))
+        nfIsRef notin node.flags and
```

Comment:
Should this also take into account `nfIsPtr`?

I believe `nfIsRef` and `nfIsPtr` should be applied to nodes which represent some value and whether that is a ref or ptr type. If that's the case then this could have issues with bad data from `opcGetImpl`, `opcGetImplTransf`, or elsewhere. Those first two don't seem to apply `nfIsRef` consistently.

I'm assuming any traversal of the sons of a `NimNode` returned by them should also be an ref, in which case I think this would fail when it shouldn't.

Might need to do a clean-up of `newNode` and co related calls in the vm as well to include `nfIsRef` and `nfIsPtr` as requireed?